controller/service.go

```diff
@@ -234,6 +264,10 @@ func getRolloutServiceKeys(rollout *v1alpha1.Rollout) []string {
 		if rollout.Spec.Strategy.BlueGreenStrategy.PreviewService != "" {
 			servicesSet[fmt.Sprintf("%s/%s", rollout.Namespace, rollout.Spec.Strategy.BlueGreenStrategy.PreviewService)] = true
 		}
+	} else if rollout.Spec.Strategy.CanaryStrategy != nil {
```

Comment:
Can we add another test to TestGetRolloutServiceKeysForCanary to cover when a canaryService specified?

Reply:
done